crates/rerun/src/web_viewer.rs

```diff
@@ -50,10 +50,43 @@ impl RemoteViewerServer {
             sender: rerun_tx,
         }
     }
+}
 
-    pub fn send(&self, msg: LogMsg) {
+impl crate::LogSink for RemoteViewerServer {
+    fn send(&mut self, msg: LogMsg) {
         if let Err(err) = self.sender.send(msg) {
             re_log::error_once!("Failed to send log message to web server: {err}");
         }
     }
 }
+
+// ----------------------------------------------------------------------------
+
+/// Extension trait for [`crate::Session`] to allow serving a web-viewer with `session.serve()`.
+pub trait WebViewerSessionExt {
+    /// Serve log-data over WebSockets and serve a Rerun web viewer over HTTP.
+    ///
+    /// If the `open_browser` argument is `true`, your default browser
+    /// will be opened with a connected web-viewer.
+    ///
+    /// If not, you can connect to this server using the `rerun` binary (`cargo install rerun`).
+    ///
+    /// NOTE: you can not connect one `Session` to another.
+    ///
+    /// This function returns immediately.
+    fn serve(&mut self, open_browser: bool);
+}
+
+impl WebViewerSessionExt for crate::Session {
+    fn serve(&mut self, open_browser: bool) {
+        if !self.is_enabled() {
+            re_log::debug!("Rerun disabled - call to serve() ignored");
+            return;
+        }
+
+        self.set_sink(Box::new(RemoteViewerServer::new(
+            self.tokio_runtime(),
+            open_browser,
+        )));
+    }
+}
```

Comment:
This should just be a free function like `spawn` and `show` are imo; the extra complexity and inconsistency brought by the extension trait doesn't seem to give any benefit?

Reply:
Probably not. I'm not sure what the best way forward is; we will likely refactor this quite a few more times.